ruby/ql/test/library-tests/ast/constants/constants.expected

```diff
@@ -65,6 +79,17 @@ constantWriteAccessQualifiedName
 | constants.rb:12:9:13:11 | ClassC | ModuleA::ModuleB::ClassC |
 | constants.rb:17:1:17:8 | GREETING | GREETING |
 | constants.rb:20:5:20:9 | Names | Names |
-| constants.rb:31:1:32:3 | ClassD | ClassD |
-| constants.rb:34:1:35:3 | ModuleC | ModuleC |
-| constants.rb:37:1:37:26 | MAX_SIZE | MAX_SIZE |
+| constants.rb:31:1:32:3 | ClassD | ModuleA::ClassD |
+| constants.rb:34:1:35:3 | ModuleC | ModuleA::ModuleC |
+| constants.rb:37:1:37:26 | MAX_SIZE | ModuleA::ModuleB::MAX_SIZE |
+| constants.rb:44:1:47:3 | ModuleB | ModuleA::ModuleB |
+| constants.rb:45:3:46:5 | ClassB | ModuleA::ModuleB::ClassB |
+| constants.rb:49:1:52:3 | ModuleA | ModuleA |
+| constants.rb:50:3:51:5 | ClassB | ModuleA::ModuleB::ClassB |
+| constants.rb:50:3:51:5 | ClassB | ModuleB::ClassB |
+| constants.rb:54:1:56:3 | Mod1 | Mod1 |
+| constants.rb:55:3:55:17 | Mod3 | Mod1::Mod3 |
+| constants.rb:58:1:61:3 | Mod4 | Mod4 |
+| constants.rb:60:3:60:23 | Mod5 | Mod1::Mod3::Mod5 |
+| constants.rb:60:3:60:23 | Mod5 | Mod4::Mod3::Mod5 |
+| constants.rb:60:3:60:23 | Mod5 | Mod3::Mod5 |
```

Comment:
I think this should actually:
```suggestion
| constants.rb:60:3:60:23 | Mod5 | Mod1::Mod3::Mod5 |
| constants.rb:60:3:60:23 | Mod5 | Mod4::Mod3::Mod5 |
```
I wonder why the `resolveConstantWriteAccess` drops `Mod4::Mod3::Mod5` instead of  `Mod3::Mod5`

Reply:
No need to look at this problem in this pull request. I think this because the implementation of `resolveConstantWriteAccess` is an over-approximation. If it were correct it should drop both `Mod3::Mod5` and `Mod4::Mod3::Mod5` .

Reply:
I think if this were correct, the possible fully qualified names for `Mod5` would be:
```ruby
Mod1::Mod3::Mod5
Mod4::Mod3::Mod5
Mod3::Mod5
```
But yeah this PR is definitely an improvement even if it's still not a perfect solution. I'm fine with merging this as-is if you are.